### ISSUE TITLE
External Connections Error about Https

### ISSUE THREAD

**Issue by deadparrotsoftware** [10up/distributor#761]
**Describe your question**
We have 2 sites with Distributor installed - one on a main domain (like **amaindomain.com**) and one on a sub-domain of the main (**sub.amaindomain.com**). 

They are virtually identical setups and both activated with our Distributor license.
Both ARE setup as https, AND HSTS etc. 
They use the same DNS on Cloudflare, are using WP 5.7.2 and identical set of plugins
And yes all the WP general settings are configured for https.
The sites don't even accept http URL's - if you use http they are reset to https.
Using Lets Encrypt SSL cert's.
On a dedicated LAMP server (Centos & Plesk) we control

So Why is Distributor refusing to setup external connections and saying our site is "NOT using https"?

Don't mean to criticize, but the Doc's (guide?) on this issue are a bit...light? I mean it tells you a code snippet you will need, but NOT what to do with it?

Sid B.

**Comment by deadparrotsoftware** [comment]
ALSO both the sites are in production/live mode

**Comment by deadparrotsoftware** [comment]
**Post UPDATE** (_Partially_ Solved)

1) This problem had NOTHING to do with our https despite the error indicating that was the cause.
It occurred because ALL 3 of our security plugins Like Wordfence consider Applications Passwords a HUGE security risk and had turned them off. _(you might want to adjust error messages from your plugin_)

2) the Distributor _plugin_ TEXT (not the docs here, the actual plugin text displays) advises you to use The "Applications Password" plugin that is now part of WP core.

3) The **connection wizard does not work** even when the external site DOES have Application Passwords ENABLED, 
admin user logged in
both running version 1.6.4 of Distributor
- it gives an ambiguous error message "Application Passwords not available"
**I WAS able to get a green dot connection doing everything manually**
(note: you should **add to your docs** the fact that you MUST use _wp-json_ in the external connection URL! even in manual mode)


**AND QUESTIONS the Distributor Docs do not answer:**

We will be using Distributor to 
1) Push Posts & Pages to sub sites (with conical links so they update etc.)
2) Have any updates/changes to those pages automatically reflected in the sub sites
Note:
The sub sites are on sub-domains of the main site, and normally will not be doing pull requests
The sub sites will NOT be pushing content anywhere, _just receiving it from the main site_.
Only Admin will have any access to Distributor - not even editors

**SO**  

1) How do we keep the enabling of  _Application Passwords_  for Distributor to work, from severely impacting security?

2) Is there a way to distribute content posts or pages WITHOUT the short codes being expanded?
(ALL the sub sites are identical setups for short codes, but values for each subsite differ. We would
like the shortcodes to use the values as they are on the external subsite, not the main site)

Sid

**Comment by jeffpaul** [comment]
> 1. How do we keep the enabling of Application Passwords for Distributor to work, from severely impacting security?

From reading through your comments it sounds like things ARE working for you, correct?  If not, can you please describe what is NOT working so I can attempt to triage that issue?


> 2. Is there a way to distribute content posts or pages WITHOUT the short codes being expanded?
(ALL the sub sites are identical setups for short codes, but values for each subsite differ. We would
like the shortcodes to use the values as they are on the external subsite, not the main site)

No, ID transformations is a topic we're looking to handle in a major Distributor update but have no immediate timeline as we're still working on scoping out that work.  If that's of a particular urgent/important nature for you, then we can discuss alternate paid support options to speed up that work for you (email me to discuss jeff.paul@10up.com).

**Comment by deadparrotsoftware** [comment]
It works as long as we do it manually. The wizard fails every time.

Your text instructions and error messages could use a bit of attention - it took a while to figure out things like what the actual problem was, based on the error message,  and to know where some of the code snippets recommended in the docs are to be used (you don’t want to ASSUME things and make changes or additions to the wrong php script :-) )

As to the other, appreciate the offer, but it’s only the shortcode preservation we are looking at - when you have a shortcode to display the site company name you want it to display the correct name for the site the page is on.  :-)

Thanks for getting back to me!

Sid

**Comment by jeffpaul** [comment]
@deadparrotsoftware do you have screenshots of where instructions were confusing?  We've spent the last couple minor releases working to fine-tune the integration with Application Passwords the plugin and Application Passwords the functionality now in WordPress core, so its possible we've glossed over some copy that makes sense to us but isn't fully clear to others.  Having those screenshot examples will help highlight places to focus in on improving.

**Comment by deadparrotsoftware** [comment]
Since we got things working I don't have screen shots of everything, but I did create a couple that show where the instructions or error messages could be improved/clarified. Hope it helps :-)

I sent them to the email address you gave me - could not get them to upload here :)

Sid

**Comment by ldollard** [comment]
The wizard doesn't work on WP 5,6(or 5.7), but it actually says that in the distributor site. Also if you read their instructions it does mention the wp-json addon. 

https://github.com/10up/distributor (scroll down for full instructions)

Wizard  mentioned here: 

> As of Distributor 1.6.0 there is an Authorization Wizard to assist in the creation of External Connections that leverages a version of Application Passwords that is bundled with Distributor versions 1.6.0 or newer. These versions will have a conflict when used on WordPress 5.6 and will require that you use the manual creation process for External Connections until a fix can be released (currently targeted for the next 1.6.2 release).

WP-JSON mentioned here:

> 5.On mainsite.com, add the External Connection URL (e.g., http://remotesite.com/wp-json). You should see a green circle and "Connection established.".

To be honest, without this team I wouldn't have been able to do the project I did as it is the ONLY single site distributor/broadcast option if you don't want to use a WP CMS.

Cheers

**Comment by deadparrotsoftware** [comment]
What I meant about the wp-json is that it doesn't say you MUST have it in
the URL
or will not work. I messed around with it a dozen times before I realized
it wasn't
a "any URL" example but HAD to end with wp-json. I thought the problem I was
having was our sub-domain, not the wp-json. Just needed a note like
*"Note that your URL for connections MUST include wp-json in the URL"*

And we have version 1.6.4 - you *could* have said it WAS that way for
1.6.4  :)

These comments were NOT intended to do anything but help. We design software
here, and have discovered the hard way that instructions and notes -
especially ones
displayed IN the plugin/app/software - can be critical, especially to
ratings

And it is nice we got it working :)


Sid


Director, Marketing & Support
Dead Parrot Software Inc.
